net_sink.go

```diff
@@ -9,10 +9,13 @@ import (
 	"strconv"
 	"sync"
 	"time"
-
-	logger "github.com/sirupsen/logrus"
 )
 
+type ErrorLogger interface {
+	Error(error)
+	Warn(error)
```

Comment:
Potential discussion point: all two callers have to construct an `error` anyway—should this accept `error`s or just a `string`?

For comparison, the standard library's logger accepts `interface{}`s and zap's accepts a `string`s and `zapcore.Field`s. 

That said, I don't think it matters all that much—either drops in perfectly anyway.

Reply:
Errors provide more ""context"" so I went with that, vs. a string.

I can be convinced since string is much more flexible, but Error is also basically a string

Reply:
Honestly fine with either, just wanted to make sure it was intentional.

Reply:
This is the actual interface: https://github.com/sirupsen/logrus/blob/master/logger.go#L181-L187

Maybe I should change it to conform to that. Seems like a totally ... fine ... interface

Reply:
Take a look now